### DISCUSSION TITLE
`create-hasura-app` and `examples` directory

### DISCUSSION THREAD

**Discussion by corysimmons** [hasura/graphql-engine#8682] [General] [answered]
Arguably the thing that made Next.js so popular was the ability to easily spin up projects via

```
$ npx create-next-app my_new_project --example some-example-name
```

They have a huge `examples` directory in their repo full of various maintained examples.

<img src="https://user-images.githubusercontent.com/224636/178292183-73cccd99-a32c-4b47-a033-2a67725ff4d6.png" width="400">

<img src="https://user-images.githubusercontent.com/224636/178292630-c8d62248-15d9-4d38-9813-21a02a532e38.png" width="400">

I would love for Hasura (and every OSS project for that matter...) to have something exactly like this.

I hate that I can't do something like...

```
$ npx create-hasura-app my_new_project --example with-next-auth
```

or

```
$ npx create-hasura-app my_new_project --example with-auto-tracked-migrations
```

**Comment by adas98012** [comment, marked as answer]
Thank you for your feedback. Meanwhile, Hasura provides an array of Sample Apps and Boilerplates for many different use cases:

https://hasura.io/docs/latest/guides/sample-apps/index/

We hope, you find this useful.  Do let us know any specific areas where you would need additional resources.
